Src/ILGPU/IR/Values/ValueKind.cs

```diff
@@ -392,8 +394,21 @@ public ValueKindAttribute(ValueKind kind)
     /// <summary>
     /// Utility methods for <see cref="ValueKind"/> enumeration values.
     /// </summary>
-    public static class ValueKinds
+    public static partial class ValueKinds
     {
+        // TODO: Source generator writes static type
+        // constructor to initialize this value, reflection free!
+        private readonly static Dictionary<ValueKind,
```

Comment:
As per the comment lines directly above, ILGPU would implement its own internal source generator to initialize a call table which maps `ValueKind` keys to their corresponding `GenericValueReader` delegates, those ofc being that which are implemented via `IValueReader` for each concrete `Value` type!